File: sort/bubbleSort.cpp
#include <iostream>
#include <vector>
using namespace std;
int main() {
  /**
   * 初始化数组
   */
  int num; // 数组个数
  cin >> num;
  int *arr = (int*)malloc(num * sizeof(int));
  for (int i = 0; i < num; ++i) {
    cin >> arr[i];
  }
  /**
   * 冒泡排序
   */
  for (int i = 0; i < num; ++i) {
    for (int j = 0; j < num - i - 1; ++j) {
      if (arr[j] > arr[j + 1]) {
        int tmp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = tmp;
      }
    }
  }
  for (int i = 0; i < num; ++i) {
    cout << arr[i];
  }
  cout << "" << endl;
  free(arr);
  return 0;
}

File: ch9排序/bubbleSort.js
// 冒泡排序
// 比较相邻的元素。如果第一个比第二个大，就交换他们两个。
// 对每一对相邻元素作同样的工作，从开始第一对到结尾的最后一对。这步做完后，最后的元素会是最大的数。
// 针对所有的元素重复以上的步骤，除了最后一个。
// 持续每次对越来越少的元素重复上面的步骤，直到没有任何一对数字需要比较。
function bubbleSort(arr) {
	for (var i = 0; i < arr.length; i++) {
		for (var j = 0; j < (arr.length - 1 - i); j++) {
			// 每冒泡一次，最后一个数字是最大的。
			if (arr[j] > arr[j + 1]) {
				[arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
			}
		}
	}
	return arr;
}
console.log(bubbleSort([3, 7, 1, 4, 9, 2]));